src/Domain/CoordinateMaps/Frustum.hpp

```diff
@@ -21,27 +21,238 @@ class er;
-/// an `OrientationMap` to the constructor. If `with_equiangular_map` is true,
-/// then this coordinate map applies a tangent function mapping to the logical
-/// xi and eta coordinates.
+/*!
+ * \ingroup CoordinateMapsGroup
+ *
+ * \brief A reorientable map from the cube to a frustum.
+ * \image html frustum_table1.png "A frustum with rectangular bases."
+ *
+ * A map from the cube to the volume determined by interpolating between two
+ * parallel rectangles each perpendicular to the \f$z\f$-axis. A Frustum map
+ * \f$\vec{x}(\xi,\eta,\zeta)\f$ is determined by specifying two
+ * heights \f$z_1 = \f$ `lower_bound` and \f$z_2 = \f$ `upper_bound` for the
+ * positions of the rectangular bases of the frustum along the \f$z-\f$axis,
+ * and the sizes of the rectangles are determined by the eight values passed to
+ * `face_vertices`:
+ *
+ * \f{align*}{
+ * &\textrm{lower x upper base} : x^{+\zeta}_{-\xi,-\eta} = x(-1,-1,1)\\
+ * &\textrm{lower x lower base} : x^{-\zeta}_{-\xi,-\eta} = x(-1,-1,-1)\\
+ * &\textrm{upper x upper base} : x^{+\zeta}_{+\xi,+\eta} = x(1,1,1)\\
+ * &\textrm{upper x lower base} : x^{-\zeta}_{+\xi,+\eta} = x(1,1,-1)\\
+ * &\textrm{lower y upper base} : y^{+\zeta}_{-\xi,-\eta} = y(-1,-1,1)\\
+ * &\textrm{lower y lower base} : y^{-\zeta}_{-\xi,-\eta} = y(-1,-1,-1)\\
+ * &\textrm{upper y upper base} : y^{+\zeta}_{+\xi,+\eta} = y(1,1,1)\\
+ * &\textrm{upper y lower base} : y^{-\zeta}_{+\xi,+\eta} = y(1,1,-1)\f}
+ *
+ * As an example, consider a frustum along the z-axis, with the lower base
+ * starting at (x,y) = (-2.0,3.0) and extending to
+ * (2.0,5.0), and with the upper base extending from (0.0,1.0) to (1.0,3.0).
+ * The corresponding value for `face_vertices` is `{{{{-2.0,3.0}}, {{2.0,5.0}},
+ * {{0.0,1.0}}, {{1.0,3.0}}}}`. The user may reorient the frustum by passing
+ * an `OrientationMap` to the constructor. If `with_equiangular_map` is true,
+ * then this coordinate map applies a tangent function mapping to the logical
+ * \f$\xi\f$ and \f$\eta\f$ coordinates. We refer to the generalized logical
+ * coordinates as \f$\Xi\f$ and \f$\mathrm{H}\f$, defined in either case as:
+ *
+ *  \f[\textrm{equiangular xi} : \Xi(\xi) = \textrm{tan}(\xi\pi/4)\f]
+ *
+ *  \f[\textrm{equiangular eta}  : \mathrm{H}(\eta) = \textrm{tan}(\eta\pi/4)\f]
+ *
+ *  With derivatives:
+ *
+ *  \f[\Xi'(\xi) = \frac{\pi}{4}(1+\Xi^2)\f]
+ *
+ *  \f[\mathrm{H}'(\eta) = \frac{\pi}{4}(1+\mathrm{H}^2)\f]
+ *
+ *  The equidistant coordinates are:
+ *
+ *  \f[ \textrm{equidistant xi}  : \Xi = \xi\f]
+ *
+ *  \f[ \textrm{equidistant eta}  : \mathrm{H} = \eta\f]
+ *
+ *  with derivatives:
+ *
+ *  \f[\Xi'(\xi) = 1\f] \f[\mathrm{H}'(\eta) = 1\f]
+ *
+ * In the case where the two
+ * rectangles are geometrically similar, the volume formed is a geometric
+ * frustum. However, this CoordinateMap generalizes to rectangles which need
+ * not be similar. In terms of the `face_vertices` variables, we define the
```

Comment:
In the past I would do this, but I've gotten comments saying that I avoid using variables that aren't defined, even in cases where they're defined immediately after. In this particular case I am also against making this change because I like having the Jacobian presented immediately after the map.